subiquity/controllers/filesystem.py

```diff
@@ -49,11 +51,64 @@
 class ProbeState(enum.IntEnum):
     NOT_STARTED = enum.auto()
     PROBING = enum.auto()
-    REPROBING = enum.auto()
     FAILED = enum.auto()
     DONE = enum.auto()
 
 
+class Probe:
+
+    def __init__(self, controller, restricted, timeout, cb):
+        self.controller = controller
+        self.restricted = restricted
+        self.timeout = timeout
+        self.cb = cb
+        self.state = ProbeState.NOT_STARTED
+        self.result = None
+
+    def start(self):
+        block_discover_log.debug(
+            "starting probe restricted=%s", self.restricted)
+        self.state = ProbeState.PROBING
+        self.controller.run_in_bg(self._bg_probe, self._probed)
+        self.controller.loop.set_alarm_in(self.timeout, self._check_timeout)
+
+    def _bg_probe(self):
+        if self.restricted:
+            probe_types = {'blockdev'}
+        else:
+            probe_types = None
+        # Should consider invoking probert in a subprocess here (so we
+        # can kill it if it gets stuck).
+        return self.controller.app.prober.get_storage(probe_types=probe_types)
+
+    def _probed(self, fut):
```

Comment:
I'm sorry, but it's not immediately clear to me what "fut" is; what type of object it is. That points me to a possible refactoring candidate (but it doesn't have to be now). Furthermore, I wonder if it really needs to be passed to the method at all; and couldn't just live in the Probe class.

Maybe it's just hard to tell by looking at the diff.

Reply:
fut is a concurrent.futures.Future and is the variable name consistently used for such a thing throughout the codebase.